experiments/FSMC.agda

```diff
@@ -0,0 +1,436 @@
+                       ≡ CompositionFreeMorphism (CompositionFreeMorphism f g) h
+   -- quotients tensor
+   FreeTensorPreserveId : ∀ {a a'} →
+                           JuxtapositionFreeMorphism
+                             (IdFreeMorphism {a}) (IdFreeMorphism {a'})
+                          ≡ IdFreeMorphism
+   FreeTensorPreserveCompose : ∀ {a a' b b' c c'}
+                                 (f : FreeMorphism a b)(f' : FreeMorphism a' b')
+                                 (g : FreeMorphism b c)(g' : FreeMorphism b' c') →
+                                   JuxtapositionFreeMorphism
+                                     (CompositionFreeMorphism f g)
+                                     (CompositionFreeMorphism f' g')
+                                 ≡ CompositionFreeMorphism
+                                     (JuxtapositionFreeMorphism f f')
+                                     (JuxtapositionFreeMorphism g g')
+   FreeTensorAssociative : ∀ {a b c d e f}
+                             (g : FreeMorphism a b)
+                             (h : FreeMorphism c d)
+                             (k : FreeMorphism e f) →
+        PathP (λ i → FreeMorphism (sym (++-assoc a c e) i) (sym (++-assoc b d f) i))
+         (JuxtapositionFreeMorphism g (JuxtapositionFreeMorphism h k))
+         (JuxtapositionFreeMorphism (JuxtapositionFreeMorphism g h) k)
+
+  FreeSymMonoidal : Category
+  obj FreeSymMonoidal = List V
+  mor FreeSymMonoidal = FreeMorphism
+  compose FreeSymMonoidal = CompositionFreeMorphism
+  identity FreeSymMonoidal = IdFreeMorphism
+  leftIdentity FreeSymMonoidal = FreeLeftidentity
+  rightIdentity FreeSymMonoidal = FreeRightidentity
+  associativity FreeSymMonoidal = FreeAssociativity
+  set-of-obj FreeSymMonoidal = isOfHLevelList 0 V-set
+
+  FreeSymMonoidal-monoidal : StrictMonoidalCategory
+  category FreeSymMonoidal-monoidal = FreeSymMonoidal
+  mapObj (tensor FreeSymMonoidal-monoidal) (a , b) = a ⊗ b
+  mapMor (tensor FreeSymMonoidal-monoidal) (f , g)
+    = JuxtapositionFreeMorphism f g
+  mapId (tensor FreeSymMonoidal-monoidal) = FreeTensorPreserveId
+  mapComp (tensor FreeSymMonoidal-monoidal) (f , f') (g , g')
+    = FreeTensorPreserveCompose f f' g g'
+  unit FreeSymMonoidal-monoidal = Unit
+  tensorIsLeftUnital FreeSymMonoidal-monoidal = refl
+  tensorIsRightUnital FreeSymMonoidal-monoidal {a} = ++-unit-r a
+  tensorIsAssociativeObj FreeSymMonoidal-monoidal {a} {b} {c}
+    = sym (++-assoc a b c)
+  tensorIsAssociativeMor FreeSymMonoidal-monoidal g h k
+    = FreeTensorAssociative g h k
+
+----- Part of the universal property
+
+module _ (V : Set)(V-set : isSet V)
+         (generatingMorphisms : List ((List V) × (List V)))
+         (ssmc : StrictMonoidalCategory)
+         (onV : V -> obj (category ssmc)) where
+
+  module Free = Category (FreeSymMonoidal V V-set generatingMorphisms)
+  module C = Category (category ssmc)
+
+  -- canonically extend onV to tensors
+  onObj : List V -> C.obj
```

Comment:
By the way, this is probably how it should be done also in the Idris codebase for the "postulated" quotient type, instead of just asking for a given `onObj` -- `onObj` also needs to preserve tensors, which is one of the things that the higher inductive type eliminator makes sure that you haven't forgotten.